primitives/src/assets.rs

```diff
@@ -107,7 +107,12 @@ where
 	type AssetRegistrarMetadata: Member + Parameter + Codec + Default + AssetMetadata;
 
 	/// The AssetLocation type: could be just a thin wrapper of MultiLocation
-	type AssetLocation: Member + Parameter + Default + TypeInfo;
+	type AssetLocation: Member
+		+ Parameter
+		+ Default
+		+ TypeInfo
+		+ From<MultiLocation>
+		+ Into<Option<MultiLocation>>;
```

Comment:
are you sure we need this ?

Reply:
So you register a asset location, how you get the para id from the generic type?
```rust
<T::AssetConfig as AssetConfig<T>>::AssetLocation
```

Reply:
Right,
Maybe we can just concretize the storage item to `MutliLocation`, or leave it like this.